### PR TITLE
Remove unnecessary variable `_hasPrimaryFocus`

### PR DESCRIPTION
**Pull request by AhmedLSayed9**
`_hasPrimaryFocus` variable and its related code is no longer needed after using `InkWell` for `DropdownButton` at https://github.com/flutter/flutter/pull/95906

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by TahaTesser** [comment]
@AhmedLSayed9 
Looks like you've made unintended with Dart format.  

Please revert the format changes and include a test for the fix.

**Review by TahaTesser — APPROVED** [review]
LGTM

**Comment by TahaTesser** [comment]
This will need a second review from @HansMuller before landing.

**Comment by QuncCccccc** [comment]
Sorry for the late response. Will take a look later today!!

File: packages/flutter/lib/src/material/dropdown.dart
Comment (QuncCccccc) [inline packages/flutter/lib/src/material/dropdown.dart:1265]: Do we also need to compare `widget.focusNode != oldWidget.focusNode` here?

Comment (AhmedLSayed9) [reply packages/flutter/lib/src/material/dropdown.dart:1265]: No.
That was needed to remove the listener from the old focusNode and add a listener to the new one.

Comment (QuncCccccc) [reply packages/flutter/lib/src/material/dropdown.dart:1265]: Oh got it. LGTM. Thanks for the contribution:)

**Review by QuncCccccc — APPROVED** [review]

**Pull request merged**
